# Question 21
-----

In [1]:
from scipy.integrate import quad
import math

class normal:
    def __init__(self, u, sigma):
        self.u = u
        self.sigma = sigma
        self.root2pi = math.sqrt((2 * math.pi)) 
    
    def normal_func(self, x):
        return (1 / (self.sigma * self.root2pi)) * math.exp((-1 * pow((x - self.u), 2)) / (2 * pow(self.sigma, 2)))

    def integrate(self, lower_limit, upper_limit):
        result, error = quad(self.normal_func, lower_limit, upper_limit)
        
        return result

    def locate(self, area):
        fx = 0
        x = -6
        while fx < area:
            fx = self.integrate(-6, x)
            x = x + 0.00001
        x = round(x, 9)
        return x

In [2]:
z = normal(0, 1)
loc_1 = z.locate(0.5)
loc_2 = z.locate(0.9)
print(loc_1)
print(loc_2)

2e-05
1.28157


$$30 - 0.000002\sigma = \mu$$

$$55 - 1.28\sigma = \mu$$

$$30 - 0.000002\sigma = 55 - 1.28\sigma$$

$$1.28\sigma = 25$$

$$\sigma = 19.53$$

$$\mu = 30$$

b) distribution is not symmetrical about the mean

In [3]:
dist = normal(30, 19.53)
prob = dist.integrate(30, 55)
expected = prob * 100
print(str(expected) + "% of employees")

39.97418376337086% of employees
